api/src/wrangler.rs

```diff
@@ -1,56 +1,92 @@
 use std::cell::RefCell;
 use std::collections::HashMap;
+use std::error::Error;
 
 use anyhow::anyhow;
+use bimap::BiBTreeMap;
+use cid::multihash::{Code, MultihashDigest};
+use cid::Cid;
+use fvm_shared::crypto::signature::{
+    Signature, SECP_PUB_LEN, SECP_SIG_LEN, SECP_SIG_MESSAGE_HASH_SIZE,
+};
+use fvm_shared::error::ExitCode;
+use fvm_shared::piece::PieceInfo;
+use fvm_shared::sector::RegisteredSealProof;
+use std::fmt;
+
+use fil_actors_runtime::runtime::builtins::Type;
+use fil_actors_runtime::runtime::{Policy, Primitives};
+use fil_actors_runtime::test_utils::{make_piece_cid, recover_secp_public_key};
 use fvm_ipld_blockstore::Blockstore;
-use fvm_ipld_encoding::{de, from_slice, RawBytes};
+use fvm_ipld_encoding::ipld_block::IpldBlock;
+use fvm_ipld_encoding::RawBytes;
 use fvm_shared::address::Address;
 use fvm_shared::bigint::Zero;
 use fvm_shared::clock::ChainEpoch;
+use fvm_shared::crypto::hash::SupportedHashes;
 use fvm_shared::econ::TokenAmount;
 use fvm_shared::message::Message;
 use fvm_shared::{ActorID, MethodNum, BLOCK_GAS_LIMIT};
 
-pub use crate::{ActorState, Bench, ExecutionResult};
+use crate::trace::InvocationTrace;
+pub use crate::{Bench, ExecutionResult};
 
-pub struct ExecutionWrangler<'b> {
-    bench: &'b mut dyn Bench,
+pub struct ExecutionWrangler<BS: Blockstore> {
```

Comment:
Is there any way we can avoid this generic? But still, this is good progress even if necessary for now.